substantial/backends/backend.py

```diff
@@ -5,20 +5,26 @@
 
 
 class Backend:
-    # run related
+    # metadata related
+    async def read_all_metadata(self, run_id: str) -> List[Metadata]:
```

Comment:
can we document the key-value structure and role of each one? I am not 100% clear about the new link ones and their actual needs after reading the PR.

Reply:
Very sorry about that, but I guess I will leave a quick summary for the time being:
The only addition on this PR is basically adding a link from workflow name to runs directly on the underlying backend, others are refactor work. It is basically just metadata so we can answer "what are all runs related to a given workflow name" (name => List run). A similar pattern is also used for child workflows on metatype (name => List name).